src/libprojplugin/libprojplugin.cpp

```diff
@@ -104,13 +146,75 @@ QString LibprojPlugin::readProjectFile()
 bool LibprojPlugin::parseMetadata(const QString & strJson)
 {
     if (strJson.isNull() || strJson.isEmpty()) {
-        qWarning() << "Nothing to parse.";
+        qWarning() << "Nothing to parse";
         return false;
     }
     else {
-        bool ok = false;
-        parsedMetadata = QtJson::parse(strJson, ok).toMap();
-        Q_ASSERT(ok);
-        return true;
+        std::string err;
+        qDebug() << "About to start parsing";
+        projectData = Json::parse(strJson.toStdString(), err);
+        if (!err.empty())
+            qWarning() << "Something wrong with parsing:\n" + QString(err.c_str());
+        return err.empty() ? true : false;
     }
 }
+
+void LibprojPlugin::triggerAddNewFileAction()
+{
+   switch(QMessageBox::question(nullptr, tr("File exists?"),tr("Are you want to add existing file (Yes) or create new (No)?"),
+                                QMessageBox::Yes, QMessageBox::No))
+   {
+   case QMessageBox::StandardButton::Yes:
+   {
+       /* TODO
+        * ...
+        */
+       break;
+   }
+   case QMessageBox::StandardButton::No:
+   {
+       QString newFilename (QFileDialog::getSaveFileName(nullptr, tr("Where you want to save new file?"), project->document()->filePath().toFileInfo().dir().path()));
+       QFile newFile (newFilename);
+       if (!newFile.exists())
+           newFile.open(QIODevice::ReadWrite | QIODevice::Text);
+       else {
+           qWarning() << "File already exists!";
```

Comment:
KISS, DRY, and encapsulate it :)
We will do it any way, when will exctract our plugin. At this stage, you can just keep it in your mind. And try to reduce depenciec between UI and business logic. Just review MVC design pattern to be little inspired in such idea.